File: chapter6/tf_tutorial/src/tf2_broadcaster.cpp
#include <ros/ros.h>
#include <tf2/LinearMath/Quaternion.h>
#include <tf2/LinearMath/Vector3.h>
#include <tf2_ros/transform_broadcaster.h>
#include <geometry_msgs/TransformStamped.h>

int main(int argc, char** argv)
{
  ros::init(argc, argv, "tf2_broadcaster");

  ros::NodeHandle n;

  tf2_ros::TransformBroadcaster tb;
  geometry_msgs::TransformStamped ts;

  ros::Rate r(1.0);
  
  while(ros::ok()){
	ts.header.stamp = ros::Time::now();
	ts.header.frame_id = "frame1";
	ts.child_frame_id = "frame2";
	
	tf2::Vector3 translation;
	translation.setValue(0, 0, 1.0);
	ts.transform.translation.x = translation.x();
	ts.transform.translation.y = translation.y();
	ts.transform.translation.z = translation.z();
	
	tf2::Quaternion rotation;
	rotation.setRPY(0, 0, 0);
	ts.transform.rotation.x = rotation.x();
	ts.transform.rotation.y = rotation.y();
	ts.transform.rotation.z = rotation.z();
	ts.transform.rotation.w = rotation.w();
	
	tb.sendTransform(ts);
	ROS_INFO("Transform Published");
	
	r.sleep();
  }
  return 0;
}

File: chapter6/tf_tutorial/scripts/tf2_broadcaster.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import rospy
import tf2_ros
import tf_conversions
from geometry_msgs.msg import TransformStamped
from geometry_msgs.msg import Vector3
from geometry_msgs.msg import Quaternion

if __name__ == '__main__':
    rospy.init_node('tf_broadcaster')

    br = tf2_ros.TransformBroadcaster()

    r = rospy.Rate(1.0)
    
    while not rospy.is_shutdown():
        t = TransformStamped()
        t.header.stamp = rospy.Time.now()
        t.header.frame_id = 'frame1'
        t.child_frame_id = 'frame2'
        
        translation = Vector3(0,0,1.0)
        t.transform.translation = translation
        
        q = tf_conversions.transformations.quaternion_from_euler(0,0,0,'sxyz')
        rotation = Quaternion(*q)
        t.transform.rotation = rotation
        
        br.sendTransform(t)
        rospy.loginfo('Transform Published')
        r.sleep()
